Source/panels/spell_book.cpp

```diff
@@ -121,62 +122,94 @@ void DrawSpellBook(const Surface &out)
+
 		if (IsValidSpell(sn) && (spl & GetSpellBitmask(sn)) != 0) {
 			spell_type st = GetSBookTrans(sn, true);
 			SetSpellTrans(st);
 			const Point spellCellPosition = GetPanelPosition(UiPanels::Spell, { 11, yp + SpellBookDescription.height });
 			DrawSpellCel(out, spellCellPosition, *pSBkIconCels, SpellITbl[sn]);
+
+			bool isSkill = false;
+			bool isStaff = false;
+			int mana = GetManaAmount(player, sn) >> 6;
+			int charges = player.InvBody[INVLOC_HAND_LEFT]._iCharges;
+			int lvl = player.GetSpellLevel(sn);
+			int baseLvl = player._pSplLvl[static_cast<std::size_t>(sn)];
+			int8_t splLvlAdd = player._pISplLvlAdd;
+			UiFlags maxSplLvlColor = baseLvl == MaxSpellLevel ? UiFlags::ColorGold : UiFlags::ColorWhite;
+			UiFlags splLvlAddColor = UiFlags::ColorWhite;
+
 			if (sn == player._pRSpell && st == player._pRSplType) {
 				SetSpellTrans(RSPLTYPE_SKILL);
 				DrawSpellCel(out, spellCellPosition, *pSBkIconCels, SPLICONLAST);
 			}
 
 			const Point line0 { 0, yp + textPaddingTop };
 			const Point line1 { 0, yp + textPaddingTop + lineHeight };
-			PrintSBookStr(out, line0, pgettext("spell", spelldata[sn].sNameText));
-			switch (GetSBookTrans(sn, false)) {
-			case RSPLTYPE_SKILL:
-				PrintSBookStr(out, line1, _("Skill"));
-				break;
-			case RSPLTYPE_CHARGES: {
-				int charges = player.InvBody[INVLOC_HAND_LEFT]._iCharges;
-				PrintSBookStr(out, line1, fmt::format(fmt::runtime(ngettext("Staff ({:d} charge)", "Staff ({:d} charges)", charges)), charges));
-			} break;
-			default: {
-				int mana = GetManaAmount(player, sn) >> 6;
-				int lvl = player.GetSpellLevel(sn);
-				PrintSBookStr(out, line0, fmt::format(fmt::runtime(pgettext(/* TRANSLATORS: UI constraints, keep short please.*/ "spellbook", "Level {:d}")), lvl), UiFlags::AlignRight);
-				if (lvl == 0) {
-					PrintSBookStr(out, line1, _("Unusable"), UiFlags::AlignRight);
+			PrintSBookStr(out, line0, pgettext("spell", spelldata[sn].sNameText), UiFlags::ColorWhite);
+
+			if (GetSBookTrans(sn, false) == RSPLTYPE_SKILL)
+				isSkill = true;
+			else if (GetSBookTrans(sn, false) == RSPLTYPE_CHARGES)
+				isStaff = true;
+
+			if (splLvlAdd > 0)
+				splLvlAddColor = UiFlags::ColorBlue;
+			else if (splLvlAdd < 0)
+				splLvlAddColor = UiFlags::ColorRed;
+
+			if (IsNoneOf(sn, SPL_REPAIR, SPL_RECHARGE, SPL_DISARM, SPL_IDENTIFY, SPL_BLODBOIL)) {
+				if (splLvlAdd != 0) {
+					PrintSBookStr(out, line0, fmt::format(fmt::runtime(pgettext(/* TRANSLATORS: UI constraints, keep short please.*/ "spellbook", "Level: {:d} ({:d})")), baseLvl, lvl), UiFlags::AlignRight | UiFlags::ColorWhite);
+					PrintSBookStr(out, line0, fmt::format(fmt::runtime(pgettext(/* TRANSLATORS: UI constraints, keep short please.*/ "spellbook", "{:d} ({:d})")), baseLvl, lvl), UiFlags::AlignRight | maxSplLvlColor);
+					PrintSBookStr(out, line0, fmt::format(fmt::runtime(pgettext(/* TRANSLATORS: UI constraints, keep short please.*/ "spellbook", "({:d})")), lvl), UiFlags::AlignRight | splLvlAddColor);

```

Comment:
Have you seen these functions?

https://github.com/diasurgical/devilutionX/blob/68d57bd2d742dd46799114dfb8a89a5888514837/Source/engine/render/text_render.hpp#L209-L229

Reply:
I haven't! Thank you. Will make another commit tomorrow night

Reply:
Do you need some help with this?